modules/arkit/arkit_interface.mm

```diff
@@ -0,0 +1,711 @@
+					projection.matrix[3][1] = m44.columns[3][1];
+					projection.matrix[0][2] = m44.columns[0][2];
+					projection.matrix[1][2] = m44.columns[1][2];
+					projection.matrix[2][2] = m44.columns[2][2];
+					projection.matrix[3][2] = m44.columns[3][2];
+					projection.matrix[0][3] = m44.columns[0][3];
+					projection.matrix[1][3] = m44.columns[1][3];
+					projection.matrix[2][3] = m44.columns[2][3];
+					projection.matrix[3][3] = m44.columns[3][3];
+				}
+			}
+		}
+	}
+}
+
+void ARKitInterface::_add_or_update_anchor(void *p_anchor) {
+	_THREAD_SAFE_METHOD_
+
+	ARAnchor *anchor = (ARAnchor *)p_anchor;
+
+	unsigned char uuid[16];
+	[anchor.identifier getUUIDBytes:uuid];
+
+	ARVRPositionalTracker *tracker = get_anchor_for_uuid(uuid);
+	if (tracker != NULL) {
+		// lets update our mesh! (using Arjens code as is for now)
+		// we should also probably limit how often we do this...
+
+		// can we safely cast this?
+		ARPlaneAnchor *planeAnchor = (ARPlaneAnchor *)anchor;
+
+		if (planeAnchor.geometry.triangleCount > 0) {
+			Ref<SurfaceTool> surftool;
+			surftool.instance();
+			surftool->begin(Mesh::PRIMITIVE_TRIANGLES);
+
+			for (int j = planeAnchor.geometry.triangleCount * 3 - 1; j >= 0; j--) {
+				int16_t index = planeAnchor.geometry.triangleIndices[j];
+				simd_float3 vrtx = planeAnchor.geometry.vertices[index];
+				simd_float2 textcoord = planeAnchor.geometry.textureCoordinates[index];
+				surftool->add_uv(Vector2(textcoord[0], textcoord[1]));
+				surftool->add_color(Color(0.8, 0.8, 0.8));
+				surftool->add_vertex(Vector3(vrtx[0], vrtx[1], vrtx[2]));
+			}
+
+			surftool->generate_normals();
+			tracker->set_mesh(surftool->commit());
+		} else {
+			Ref<Mesh> nomesh;
+			tracker->set_mesh(nomesh);
+		}
+
+		// Note, this also contains a scale factor which gives us an idea of the size of the anchor
+		// We may extract that in our ARVRAnchor class
+		Basis b;
+		matrix_float4x4 m44 = anchor.transform;
+		b.elements[0].x = m44.columns[0][0];
+		b.elements[1].x = m44.columns[0][1];
+		b.elements[2].x = m44.columns[0][2];
+		b.elements[0].y = m44.columns[1][0];
+		b.elements[1].y = m44.columns[1][1];
+		b.elements[2].y = m44.columns[1][2];
+		b.elements[0].z = m44.columns[2][0];
+		b.elements[1].z = m44.columns[2][1];
+		b.elements[2].z = m44.columns[2][2];
+		tracker->set_orientation(b);
+		tracker->set_rw_position(Vector3(m44.columns[3][0], m44.columns[3][1], m44.columns[3][2]));
+	}
+}
+
+void ARKitInterface::_remove_anchor(void *p_anchor) {
+	_THREAD_SAFE_METHOD_
+
+	ARAnchor *anchor = (ARAnchor *)p_anchor;
+
+	unsigned char uuid[16];
+	[anchor.identifier getUUIDBytes:uuid];
+
+	remove_anchor_for_uuid(uuid);
+}
+
+ARKitInterface::ARKitInterface() {
+	initialized = false;
+	session_was_started = false;
+	plane_detection_is_enabled = false;
+	light_estimation_is_enabled = false;
+	ar_session = NULL;
+	z_near = 0.01;
+	z_far = 1000.0;
+	projection.set_perspective(60.0, 1.0, z_near, z_far, false);
```

Comment:
Should this always be 60 FOV?

Reply:
it's just a default, it gets overwritten in process by the projection matrix ARKit provides us with.